app/views/spree/orders/_line_item.html.haml

```diff
@@ -22,7 +22,8 @@
   %td.text-right.cart-item-price{"data-hook" => "cart_item_price"}
     = line_item.single_display_amount_with_adjustments.to_html
   %td.text-center.cart-item-quantity{"data-hook" => "cart_item_quantity"}
-    = item_form.number_field :quantity, :min => 0, "ofn-on-hand" => variant.on_hand, "ng-model" => "line_item_#{line_item.id}", :class => "line_item_quantity", :size => 5
+    - finalized_quantity = @order.completed? ? line_item.quantity : 0
```

Comment:
Is the line_item.quantity column nullable? I'd like to trust its value to be an integer because we'll end up with lots of defensive programming otherwise.

Reply:
I am not sure I understand your comment.
do you say that because you think this ternary op is defensive programming?
this is not defensive programming, this is calculating finalized_quantity (stock already taken from stock levels) and the finalized_quantity is not line_item.quantity if order is not complete because the stock was not finalized yet (taken from stock levels). do you see the point or do you mean something else?

Reply:
maybe I need to add a comment to the code explaining this?

Reply:
I guess that he meant if `line_item.quantity` can be nil, we need to add more defensive programming here (and return 0 instead). But I just checked the schema and `quantity` can't be null in the database. So it shouldn't be nil here either.

Reply:
yes, that's my point. I wanted to be sure we are not defending against `line_item.quantity` being nil. That's a DB responsibility in this case.

Reply:
ok, thanks for clarifying. we are not defending here and we dont need to :-)